openvpn.c

```diff
@@ -880,18 +880,79 @@ GenericPassDialogFunc(HWND hwndDlg, UINT msg, WPARAM wParam, LPARAM lParam)
                     SetWindowPos(hwndDlg, NULL, 0, 0, rect.right, rect.bottom, SWP_NOMOVE);
                     PrintDebug(L"Window resized to = %d %d", rect.right, rect.bottom);
                 }
+                ShowWindow(GetDlgItem(hwndDlg, ID_CHK_SAVE_PASS), SW_HIDE);
             }
             else if (param->flags & FLAG_PASS_TOKEN)
             {
                 SetWindowText(hwndDlg, LoadLocalizedString(IDS_NFO_TOKEN_PASSWORD_CAPTION));
                 SetDlgItemText(hwndDlg,
                                ID_TXT_DESCRIPTION,
                                LoadLocalizedString(IDS_NFO_TOKEN_PASSWORD_REQUEST, param->id));
+
+                if (param->c->flags & FLAG_DISABLE_SAVE_PASS) 
+                {
+                    ShowWindow(GetDlgItem(hwndDlg, ID_CHK_SAVE_PASS), SW_HIDE);
```

Comment:
Isn't it better to make this hiding of the save password checkbox the default outside of all if-else clauses? That will simplify the logic and have less deeply nested code.
Like, the following line could become `else if (!(param->c->flags & FLAG_DISABLE_SAVE_PASS) && RecallSmart...())`